### ISSUE TITLE
Padding below 'change default skin tone' increases on no result found in emoji picker

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15868]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. Open any report
3. Open emoji picker
4. Search for any emoji eg: 'skull' and observe padding below 'change default skin tone'
5. Search for emoji that does not exist eg: 'skulll' and observe padding below 'change default skin tone'

## Expected Result:
Padding below 'change default skin tone' should be same all the time

## Actual Result:
Padding below 'change default skin tone' increases on no result found search in emoji picker

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.82-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/224448719-454426eb-6da8-4d43-9ef3-3af32f0d148e.mp4

https://user-images.githubusercontent.com/43996225/224448728-7740762b-da80-4bed-a78b-aa7e23a781ce.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1678467171785959

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @anmurali (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@anmurali Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~019f078af8705c66b2

**Comment by MelvinBot** [comment]
Current assignee @anmurali is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @rushatgabhane (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @iwiznia (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by PrashantMangukiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Padding below 'change default skin tone' increases on no result found in emoji picker.

### What is the root cause of that problem?
It is rendering "No results found" message via `<Text>` component at line 501 
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/components/EmojiPicker/EmojiPickerMenu/index.js#L499-L513

We can see that `styles.dFlex` is there but `styles.flex1` missing, so whenever "No results found" message render at that time it will not occupy all space and so due to that `<EmojiSkinToneList>` component (it is below that) slightly jump up words.

### What changes do you think we should make in order to solve the problem?
Within `src/components/EmojiPicker/EmojiPickerMenu/index.js` we have to add `styles.flex1` for "No results found" Text component as shown below, It will solve the problem. I tested it locally, It works as expected and solve this issue.

```javascript
<Text
  style={[
      styles.disabledText,
      styles.emojiPickerList,
      styles.dFlex,
      styles.flex1,   // *** ADD THIS ***
      styles.alignItemsCenter,
      styles.justifyContentCenter,
      this.isMobileLandscape() && styles.emojiPickerListLandscape,
  ]}
>
    {this.props.translate('common.noResultsFound')}
</Text>
```

### What alternative solutions did you explore? (Optional)

None

**Comment by abdulrahuman5196** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
This issue(https://github.com/Expensify/App/issues/15868) seems to be a regression from https://github.com/Expensify/App/pull/15730

### What is the root cause of that problem?

We have increased the height of the emoji component from 392 to 416 i.e 416-392 = 24 new space
And increased skin tone padding from p1 to p3 i.e 16 space used
But we have not updated the height of the 'No results' component similarly here - https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerMenu/index.js#L501 leading to 8 unused space jumping up when we show no results component

### What changes do you think we should make in order to solve the problem?

Option 1: Currently we show the 'no results component' for only big screen web devices in which case the emoji picker component has 416 height. So instead of hardcoding height of 'No results component', we can add styles.h100(100% height) here - https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPickerMenu/index.js#L507. This way 'No results' component will automatically adjust to the height of emoji picker and we don't need to update the height value of 'No results' component everytime we update the picker height.
Option 2: We can hardcode the height as 296(currently 288 +  8 extra left space) for the 'No result component'.

### What alternative solutions did you explore? (Optional)
There are multiple ways to achieve the result by modifying different styles but the above seems the best and future proof.

**Comment by stitesExpensify** [comment]
FYI to contributors there has already been a proposal to fix this regression by @abdulrahuman5196 here, which will be prioritized https://github.com/Expensify/App/pull/15730#issuecomment-1470509444

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Height inconsistency for Not Result Found in Emoji Picker

### What is the root cause of that problem?
Incorrect height is given for the `noResultsFound` Text, It is given currently `288px` which should be `256px(for 8 rows of emoji flatlist) + 40px(for CategoryShortcutBar)`
https://github.com/Expensify/App/blob/20d5fdc373292621311288493eb994b644ef0ac8/src/components/EmojiPicker/EmojiPickerMenu/index.js#L504
https://github.com/Expensify/App/blob/20d5fdc373292621311288493eb994b644ef0ac8/src/styles/styles.js#L1497-L1498

### What changes do you think we should make in order to solve the problem?
We can add style `styles.flex1` here which will grow with the available height for the notFound Text div element.
https://github.com/Expensify/App/blob/8a04b2c9169a9294011535ce86b4c12018f7fa11/src/components/EmojiPicker/EmojiPickerMenu/index.js#L505

**Extra:** For Web we need `256px(for 8 rows * 32)` height for the emoji flatlist and currently we have `288px`, in native we are showing 9 rows where it would be required to `288px`. So either we can maintain separate class for web & native OR in native also we can show 8 rows which we can achieve by general change of height to `256px` in class `emojiPickerList`.

**Comment by MelvinBot** [comment]
@iwiznia, @anmurali, @rushatgabhane Eep! 4 days overdue now. Issues have feelings too...

**Comment by MelvinBot** [comment]
@iwiznia, @anmurali, @rushatgabhane 8 days overdue is a lot. Should this be a Weekly issue? If so, feel free to change it!

**Comment by anmurali** [comment]
Posted [here](https://expensify.slack.com/archives/C049HHMV9SM/p1679339807786249?thread_ts=1678467171.785959&cid=C049HHMV9SM) to make a decision here

**Comment by abdulrahuman5196** [comment]
@rushatgabhane  Thank you for the review. I have updated the original proposal https://github.com/Expensify/App/issues/15868#issuecomment-1471979737

>That will break the emoji screen when we reduce the the screen size.

I think we should not show the 'No results' component in this case as there is no search bar. This would be a different bug and We should create a separate issue for this.

Even though I think the above break is a different bug scenerio, I think we shouldn't break current experience. So I have extended my proposal with an option 3 on why this break is happening and its potential long term fixes with respect to our current bug. Could you kindly have a look into it here - https://github.com/Expensify/App/issues/15868#issuecomment-1471979737?


>I like option 2 the best.
>But I have a concern - What if we change the padding again in future. Is there a solution that will auto adjust the height?

Yes. I have called out the same in option 3.

**Comment by iwiznia** [comment]
Cool, agree. @anmurali let's hire @tienifr

**Comment by Pujan92** [comment]
@rushatgabhane @iwiznia  can you plz review my [proposal](https://github.com/Expensify/App/issues/15868#issuecomment-1471985688) too once?

**Comment by MelvinBot** [comment]
@iwiznia @anmurali @rushatgabhane this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by tienifr** [comment]
@rushatgabhane Could I raise a PR now?

**Comment by MelvinBot** [comment]
📣 @tienifr You have been assigned to this job by @stitesExpensify!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by tienifr** [comment]
Hi @rushatgabhane @iwiznia   [the PR](https://github.com/Expensify/App/pull/16439) is ready for review

**Comment by tienifr** [comment]
# Regression Test Proposal

**Bug**: Padding below 'change default skin tone' increases on no result found in emoji picker 

**Proposed Test Steps**:

1. Open the app
2. Open any report
3. Open emoji picker
4. Search for emoji that does not exist eg: 'skulll' 
5. Verify that the position of 'change default skin tone' should be the same all the time


Do we :+1: or :-1:

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.92-2 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/16439


If no regressions arise, payment will be issued on 2023-04-07. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@rushatgabhane / @iwiznia] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@rushatgabhane / @iwiznia] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@rushatgabhane / @iwiznia] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@anmurali] Determine if we should create a regression test for this bug.
- [ ] [@rushatgabhane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@anmurali] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by dhanashree-sawant** [comment]
Hi @rushatgabhane, @iwiznia , @anmurali , ping for payment.

**Comment by MelvinBot** [comment]
@iwiznia, @anmurali, @rushatgabhane, @tienifr Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@iwiznia, @anmurali, @rushatgabhane, @tienifr Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by anmurali** [comment]
@rushatgabhane can you take care of your items in the checklist? I think I pay and add regression tests last.

**Comment by iwiznia** [comment]
So what is missing here? Payment with the regression taken into consideration @anmurali?

**Comment by anmurali** [comment]
@tienifr - I cannot find you on Upwork. Your Slack and GH both don't list your name. Can you post it here so I know how to find you to pay you?

**Comment by iwiznia** [comment]
Bump @tienifr

**Comment by tienifr** [comment]
sorry missed this, @anmurali here's my Upwork profile https://www.upwork.com/freelancers/~01991fd5e5c11ef3ba, thanks

**Comment by anmurali** [comment]
Offers sent

**Comment by anmurali** [comment]
Everyone's paid now!

**Comment by tienifr** [comment]
> hi @anmurali The PR is merged within 3 days of assignment so unless I'm missing something, this should be eligible for the speed bonus 🙇‍♂️.

@anmurali can you help take a look? Thanks!

**Comment by tienifr** [comment]
> hi @anmurali The PR is merged within 3 days of assignment so unless I'm missing something, this should be eligible for the speed bonus 🙇‍♂️.

@anmurali could you help to take a look at [this](https://github.com/Expensify/App/issues/15868#issuecomment-1531621295)?

@iwiznia just wondering if you can help here if @anmurali is unavailable? 

Thanks!

**Comment by tienifr** [comment]
Thanks!

> Done. @tienifr if you have asked her once and haven't received an answer in a week perhaps you can ping in Slack and tag the person you are waiting on instead?

@anmurali sure I'll take note of that, sorry for any inconvenience 🙇
